reconcile/aws_ami_cleanup/integration.py

```diff
@@ -211,11 +211,18 @@ def run(dry_run: bool, thread_pool_size: int, defer: Optional[Callable] = None)
     # cannot still be properly monitored (see https://issues.redhat.com/browse/APPSRE-7674),
     # it's easy that it breaks without being noticed. Once it is properly monitored, this should
     # be moved to a typed query.
-    cleanup_accounts = [
-        a
-        for a in queries.get_aws_accounts(terraform_state=True, cleanup=True)
-        if a.get("cleanup")
-    ]
+    query_data = queries.get_aws_accounts(terraform_state=True, cleanup=True)
+    cleanup_accounts = []
+    for data in query_data:
+        cleanups = data.get("cleanup")
+        if not cleanups:
+            continue
+        is_ami_related = False
+        for cleanup in cleanups:
+            is_ami_related |= cleanup.get("provider") == "ami"
+        if not is_ami_related:
+            continue
+        cleanup_accounts.append(data)
```

Comment:
Since the filter logic become complex, I suggest to extract it.


```python
def _is_cleanup_account(account: Mapping[str, Any]) -> bool:
    cleanups = account.get("cleanup")
    if not cleanups:
        return False
    return any(cleanup.get("provider") == "ami" for cleanup in cleanups)
```

Then it can be used like

```python
    cleanup_accounts = list(
        filter(
            _is_cleanup_account,
            queries.get_aws_accounts(terraform_state=True, cleanup=True),
        )
    )
```

Plus we need a new test for this

Reply:
Just trying to quick fix the open alert for now - but I fully agree we should add tests here :+1:

Reply:
Then we should add a proper fix after this, a quick fix without tests is a new bug.

Reply:
@fishi0x01 Can you create a JIRA card to track the refactor so that we don't lose track of it?